query/src/main/scala/filodb/query/exec/rangefn/RangeInstantFunctions.scala

```diff
@@ -122,17 +122,15 @@ class ResetsFunction extends RangeFunction {
 
   def addedToWindow(row: TransientRow, window: Window): Unit = {
     val size = window.size
+    if (row.value.isNaN && !resets.isNaN) resets += 1
```

Comment:
(10, NaN, NaN, 5) would be counted as two because of this. Best is to simply treat NaN as 0 when comparing against previous or next value.

Reply:
true, two adjacent NaN's should not increment `resets`

Reply:
Made the change and added a test case. Thanks for letting me know!